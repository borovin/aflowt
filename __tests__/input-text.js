const test = require('ava');
const Browser = require('./helpers/browser');
const {prettyPrint} = require('html');

test.beforeEach(t => {
    t.context.browser = new Browser();

    return t.context.browser.goto('/examples/input-text.html');
});

test.afterEach(t => {
    return t.context.browser.close();
});

test('Default text input', async t => {
    const snapshot = await t.context.browser.snapshot('b-input-text:nth-child(1)');

    t.snapshot(prettyPrint(snapshot));
});

test('Floating label input', async t => {
    const snapshot = await t.context.browser.snapshot('b-input-text:nth-child(2)');

    t.snapshot(prettyPrint(snapshot));
});

test('Filled input', async t => {
    const snapshot = await t.context.browser.snapshot('b-input-text:nth-child(3)');

    t.snapshot(prettyPrint(snapshot));
});

test('Input error', async t => {
    const snapshot = await t.context.browser.snapshot('b-input-text:nth-child(4)');

    t.snapshot(prettyPrint(snapshot));
});

test('Input password', async t => {
    const snapshot = await t.context.browser.snapshot('b-input-text:nth-child(5)');

    t.snapshot(prettyPrint(snapshot));
});