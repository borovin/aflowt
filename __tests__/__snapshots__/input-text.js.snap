// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default text input 1`] = `
"<b-input-text name=\\"a\\" placeholder=\\"Input Placeholder\\" type=\\"text\\">
    <input type=\\"text\\" placeholder=\\"Input Placeholder\\" name=\\"a\\">
    <b-input-text--border></b-input-text--border>
</b-input-text>"
`;

exports[`Filled input 1`] = `
"<b-input-text name=\\"c\\" value=\\"Example Value\\" label=\\"Input Label\\" type=\\"text\\">
    <input value=\\"Example Value\\" type=\\"text\\" name=\\"c\\">
    <b-input-text--border></b-input-text--border>
    <b-input-text--label>Input Label</b-input-text--label>
</b-input-text>"
`;

exports[`Floating label input 1`] = `
"<b-input-text name=\\"b\\" label=\\"Floating Label\\" type=\\"text\\">
    <input type=\\"text\\" name=\\"b\\">
    <b-input-text--border></b-input-text--border>
    <b-input-text--label>Floating Label</b-input-text--label>
</b-input-text>"
`;

exports[`Input error 1`] = `
"<b-input-text name=\\"d\\" value=\\"Example Value\\" label=\\"Input Label\\" error=\\"some input error\\"
type=\\"text\\">
    <input value=\\"Example Value\\" type=\\"text\\" name=\\"d\\">
    <b-input-text--border></b-input-text--border>
    <b-input-text--label>Input Label</b-input-text--label>
</b-input-text>"
`;

exports[`Input password 1`] = `
"<b-input-text name=\\"e\\" type=\\"password\\" value=\\"password\\" label=\\"Password\\">
    <input value=\\"password\\" type=\\"password\\" name=\\"e\\">
    <b-input-text--border></b-input-text--border>
    <b-input-text--label>Password</b-input-text--label>
</b-input-text>"
`;
